### PR TITLE
Create dashboard and dashboard presenter class

### PR DESCRIPTION
**Pull request by meganft**
@jmejia 
Made the initial dashboard and extracted some of the login into a dashboard presenter. I am planning to pull more of the logic out of the views and most likely render it as a partial as it get more complex, but is this the direction you think will work best for this?

### CODE REVIEW

File: app/models/dashboard_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class DashboardPresenter
+
+  attr_reader :events
+
+  def initialize(user)
+    @events = user.events
+  end
```
Comment (jmejia) [inline app/models/dashboard_presenter.rb:8]: @meganft Thank you for not using `current_user` here! 🎉 🌮

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class User < ApplicationRecord
   has_secure_password
 
+  has_many :user_events
   has_many :events, through: :user_events
   has_many :roles, through: :user_events
-  has_many :user_events
```
Comment (jmejia) [inline app/models/user.rb:6]: Nice :)

File: app/views/dashboard/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div class="container">
+  <h2>Welcome <%= current_user.first_name %></h2>
+  
+  <h4>Here are your current events:</h4>
+
+  <% @presenter.events.each do |event| %>
+    <h5><%= event.name %></h5>
+    <% event.roles.each do |role| %>
+      </p>Your role is: <%= role.name %></p>
```
Comment (jmejia) [inline app/views/dashboard/show.html.erb:9]: @meganft If they have more than one role I wonder if doing something like this makes sense:

```ruby
</p>Your role(s): <%= roles.pluck(:name).join(", ") %></p>
```

File: spec/features/guests/guest_can_create_an_account_spec.rb [hunk 4]
```diff
@@ -29,9 +28,27 @@
...
+      fill_in "user[first_name]", with: user.first_name
+      fill_in "user[last_name]", with: user.last_name
+      fill_in "user[street_number]", with: user.street_number
+      fill_in "user[unit_number]", with: "45"
+      fill_in "user[city]", with: user.city
+      fill_in "user[state]", with: user.state
+      fill_in "user[password]", with: user.password
+      fill_in "user[password_confirmation]", with: user.password
+      fill_in "user[picture]", with: user.picture
+      # select("#{cat_1.name}", from: 'user[category_id]')
```
Comment (jmejia) [inline spec/features/guests/guest_can_create_an_account_spec.rb]: Can we get rid of the commented out test code?

File: app/models/dashboard_presenter.rb (hunk 1 above)
Comment (meganft) [reply app/models/dashboard_presenter.rb:8]: @jmejia 👍 🌯

**Review by jmejia — COMMENTED** [review]
Nice work! I left a few small adjustments here but your overall approach is spot on 👍

File: spec/features/guests/guest_can_create_an_account_spec.rb (hunk 4 above)
Comment (meganft) [reply spec/features/guests/guest_can_create_an_account_spec.rb]: @jmejia Just got rid of it!

File: app/views/dashboard/show.html.erb (hunk 3 above)
Comment (meganft) [reply app/views/dashboard/show.html.erb:9]: @jmejia 
I'll give this a try!

**Comment by annadolan** [comment]
💯

**Pull request merged**